test/Microsoft.Azure.WebJobs.Host.EndToEndTests/AsyncChainEndToEndTests.cs

```diff
@@ -222,6 +224,42 @@ public void FunctionFailures_LogsExpectedTraceEvent()
         }
 
         [Fact]
+        public void RandGuidOutput_GeneratesRandomIDs()
+        {
+            JobHost host = new JobHost(_hostConfig);
+
+            var blobClient = _fixture.StorageAccount.CreateCloudBlobClient();
+            var container = blobClient.GetContainerReference("test-output");
+            if (container.Exists())
+            {
+                foreach (CloudBlockBlob blob in container.ListBlobs())
+                {
+                    blob.Delete();
+                }
+            }
+
+            MethodInfo methodInfo = GetType().GetMethod("RandGuidOutput");
+            for (int i = 0; i < 3; i++)
+            {
+                var arguments = new Dictionary<string, object>
+                {
+                    { "input", i.ToString() }
+                };
+                host.Call(methodInfo, arguments);
+            }
+
+            // We expect 3 separate blobs to have been written
+            var blobs = container.ListBlobs().Cast<CloudBlockBlob>().ToArray();
+            Assert.Equal(3, blobs.Length);
+            foreach (var blob in blobs)
+            {
+                string content = blob.DownloadText(Encoding.UTF8);
+                int blobInt = int.Parse(content.Trim(new char[] { '\uFEFF', '\u200B' }));
```

Comment:
I'm suggesting you check the *name* of the blob. Isn't that what you're using {rand-guid} to replace? The content of the blob is an int, but the name should be a guid.

---
In reply to: [70699572](https://github.com/Azure/azure-webjobs-sdk/pull/767#discussion_r70699572) [](ancestors = 70699572)